### PR TITLE
Add support for tagged literals

### PR DESCRIPTION
**Pull request by moxaj**

### CODE REVIEW

File: src/cljs/snapshot/lumo/repl.cljs [hunk 1]
```diff
@@ -902,6 +902,35 @@
           (recur (rt/read-char reader)))
         (str sb)))))
 
+(defn get-data-readers*
+  "Returns the merged data reader mappings."
+  []
+  (reduce (fn [data-readers url+source]
+            (let [url      (aget url+source "url")
```
Comment (anmonteiro) [inline src/cljs/snapshot/lumo/repl.cljs]: please use `(.-url url+source)`. `aget` is only meant to be used on arrays.

File: src/cljs/snapshot/lumo/repl.cljs [hunk 2]
```diff
@@ -902,6 +902,35 @@
...
+                  mappings (r/read-string source)]
+              (when-not (map? mappings)
+                (throw (ex-info (str "Not a valid data-reader map")
+                                {:url url})))
+              (reduce-kv (fn [data-readers tag fn-sym]
+                           (when-not (symbol? tag)
+                             (throw (ex-info (str "Invalid form in data-reader file")
+                                             {:url  url
+                                              :form tag})))
+                           (when (and (data-readers tag)
```
Comment (anmonteiro) [inline src/cljs/snapshot/lumo/repl.cljs]: please use `get` to make it explicit for readers

File: src/cljs/snapshot/lumo/repl.cljs [hunk 3]
```diff
@@ -902,6 +902,35 @@
...
+                                      (not= (mappings tag) fn-sym))
+                                (throw (ex-info "Conflicting data-reader mapping"
+                                         {:url      url
+                                          :conflict tag
+                                          :mappings data-readers})))
+                           (assoc data-readers tag (eval fn-sym)))
```
Comment (anmonteiro) [inline src/cljs/snapshot/lumo/repl.cljs]: I'm unsure whether we should also `require` the `fn-sym` namespace or assume the user has done it before. 
Currently JVM ClojureScript expects users of data readers to have required the namespaces where they are declared so I'm inclined to say this is fine as is.

Comment (moxaj) [inline src/cljs/snapshot/lumo/repl.cljs]: I've left it as is for now. You probably don't want spoiled lumo users to complain when using regular cljs :)

**Pull request merged**